tasks/tasks.go

```diff
@@ -0,0 +1,93 @@
+package tasks
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// Status is the status of the task.
+type Status int
+
+// TaskExecutor executes a series of task functions.
+type TaskExecutor interface {
+	QueueTask(task func() error) string
+	GetTaskStatus(id string) (Status, error)
+	completeTask(id string, task func() error)
+}
+
+type defaultExecuter struct {
+	sync.RWMutex
+	m map[string]Status
+}
+
+// These are some enumerated Status constants.
+// INPROGRESS: Task is still in progress.
+// SUCCESS: Task finished successfully.
+// FAILURE: Task finished unsuccessfully.
+// NOTFOUND: Task could not be found.
+const (
+	INPROGRESS Status = iota
+	SUCCESS
+	FAILURE
+	NOTFOUND
+)
+
+// NewTaskExectuer returns a TaskExecutor ready to execute.
+func NewTaskExectuer() TaskExecutor {
+	return &defaultExecuter{m: make(map[string]Status)}
+}
+
+// QueueTask initializes a new task. It takes a generic task function.
+func (ex *defaultExecuter) QueueTask(task func() error) string {
+	id := generateID(20)
+	ex.Lock()
+	ex.m[id] = INPROGRESS
+	ex.Unlock()
+	go ex.completeTask(id, task)
+	return id
+}
+
+// GetTaskStatus gets the current status of a task.
+func (ex *defaultExecuter) GetTaskStatus(id string) (Status, error) {
+	ex.RLock()
+	defer ex.RUnlock()
+
+	if status, ok := ex.m[id]; ok {
+		return status, nil
+	}
+	return NOTFOUND, errors.New("Invalid id")
```

Comment:
Does it make any sense to return an error here?. Let's say that I want to query for a task and check it's status
```go
status, err := executor.GetTaskStatus("123456....")
if err != nil && status == tasks.SUCCESS { // Why do I need to check for errors here if I will still check for the status code?
....
}

```